swift-sdk/Internal/InAppManager+Functions.swift

```diff
@@ -52,25 +52,26 @@ struct MessagesProcessor {
         
         ITBDebug("processing message with id: \(message.messageId)")
         
-        if inAppDisplayChecker.isOkToShowNow(message: message) {
-            ITBDebug("isOkToShowNow")
-            if inAppDelegate.onNew(message: message) == .show {
-                ITBDebug("delegate returned show")
-                return .show(message)
-            } else {
-                ITBDebug("delegate returned skip")
-                return .skip(message)
-            }
-        } else {
+        guard inAppDisplayChecker.isOkToShowNow(message: message) else {
             ITBDebug("Not ok to show now")
-            
             return .wait
         }
+        
+        ITBDebug("isOkToShowNow")
+        
+        if inAppDelegate.onNew(message: message) == .show {
+            ITBDebug("delegate returned show")
+            return .show(message)
+        } else {
+            ITBDebug("delegate returned skip")
+            return .skip(message)
+        }
     }
     
     private func getFirstProcessableTriggeredMessage() -> IterableInAppMessage? {
         messagesMap.values
             .filter(MessagesProcessor.isProcessableTriggeredMessage)
+            .filter({ !$0.read })
```

Comment:
You can add this to `MessagesProcessor.isProcessableTriggeredMessage` method. Not need to do filtering twice.